src/pydap/lib.py

```diff
@@ -120,14 +119,30 @@ def quote(name):
     safe = '%_!~*\'-"'
     return quote_(name.encode('utf-8'), safe=safe).replace('.', '%2E')
 
+def quote_val(name):
+    """Return quoted name according to the DAP specification.
+       Used for attributes and values.
+       Encode every cahracter which is not 7bis ASCII.
+
+        >>> quote_val("White space. /;")
+        'White space. /;'
+        >>> quote_val("Period.")
+        'Period.'
+
+    """
+    safe = ''.join([chr(x) for x in range(128)])
+    return quote_(name.encode('utf-8'), safe=safe)
 
 def encode(obj):
     """Return an object encoded to its DAP representation."""
+    try:
+        return quote_val(obj)
+    except:
```

Comment:
Which exception is expected here? You should probably except only that one here.

Reply:
Echo'ing @laliberte, @pauleckhardt can we be more explicit here?